From a416220b783e5408f119ba7ba9b1be6d3c5a7da6 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Mon, 12 Feb 2024 17:41:14 +0100
Subject: [PATCH 0786/1695] plat-stm32mp1: stm32mp1_pmic: add regulator
 configuration

Configure the default ramp delay for the STPMIC1 regulators, the
DT configuration should be use only to change this values, for example
fort board or external device constraints.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e677d28c5ed565e5f51ee320eed7d14dbf121871 plat-stm32mp1: stm32mp1_pmic: add regulator configuration]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I5c577e10570d7ecae48ff955ffb4d94ba897ada1
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/359383
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 .../arm/plat-stm32mp1/drivers/stm32mp1_pmic.c | 28 ++++++++++++++++---
 1 file changed, 24 insertions(+), 4 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c b/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c
index c7404c7d1..b5517b07d 100644
--- a/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c
+++ b/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c
@@ -526,8 +526,26 @@ static TEE_Result pmic_regu_pm(enum pm_op op, uint32_t pm_hint __unused,
 	return TEE_SUCCESS;
 }
 
-static TEE_Result pmic_regu_init(struct regulator *regulator,
-				 const void *fdt __unused, int node __unused)
+static TEE_Result pmic_regu_init(struct regulator *regulator)
+{
+	/* Default configuration for STPMIC regulators */
+	regulator->ramp_delay_uv_per_us = U(2200);
+	regulator->enable_ramp_delay_us = U(1000);
+
+	return TEE_SUCCESS;
+}
+
+static TEE_Result pmic_sw_init(struct regulator *regulator)
+{
+	/* Default configuration for STPMIC power switch */
+	regulator->enable_ramp_delay_us = U(1000);
+
+	return TEE_SUCCESS;
+}
+
+static TEE_Result pmic_supplied_init(struct regulator *regulator,
+				     const void *fdt __unused,
+				     int node __unused)
 {
 	struct pmic_regulator_data *priv = regulator->priv;
 	struct stpmic1_bo_cfg cfg = { };
@@ -567,14 +585,16 @@ static const struct regulator_ops pmic_regu_ops = {
 	.set_voltage = pmic_set_voltage,
 	.get_voltage = pmic_get_voltage,
 	.supported_voltages = pmic_list_voltages,
-	.supplied_init = pmic_regu_init,
+	.init = pmic_regu_init,
+	.supplied_init = pmic_supplied_init,
 };
 DECLARE_KEEP_PAGER(pmic_regu_ops);
 
 static const struct regulator_ops pmic_sw_ops = {
 	.set_state = pmic_set_state,
 	.get_state = pmic_get_state,
-	.supplied_init = pmic_regu_init,
+	.init = pmic_sw_init,
+	.supplied_init = pmic_supplied_init,
 };
 DECLARE_KEEP_PAGER(pmic_sw_ops);
 
-- 
2.47.2

